account_balance_reporting/models/account_balance_reporting_report.py

```diff
@@ -206,19 +206,19 @@ def name_get(self):
             res.append((item.id, "[%s] %s" % (item.code, item.name)))
         return res
 
-    def name_search(self, cr, uid, name, args=None, operator='ilike',
-                    context=None, limit=80):
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
         """Redefine the method to allow searching by code."""
-        ids = []
         if args is None:
             args = []
-        if name:
-            ids = self.search(cr, uid, [('code', 'ilike', name)] + args,
-                              limit=limit, context=context)
-        if not ids:
-            ids = self.search(cr, uid, [('name', operator, name)] + args,
-                              limit=limit, context=context)
-        return self.name_get(cr, uid, ids, context=context)
+        return self.search(
+            [
+                '|',
+                ('code', operator, name),
+                ('name', operator, name),
+            ] + args,
+            limit=limit
+        ).name_get()
```

Comment:
`args` es un dominio adicional, pero que no espera encontrar argumentos con `|`. Todo lo esperado deben ser tuplas para AND. Si pongo el código que dices, tendría que ser:
```python
args += [('|', ('code', operator, name)]
return super(...).name_search(..., args, ...)
```
Y efectivamente con la implementación actual el dominio que me aplicaría sería:
```python
[('|', ('code', operator, name), (self._rec_name, operator, name)]
```
Pero esto tiene muchas  pegas:
* Mañana Odoo cambia la implementación por lo que sea y pone el orden de la suma al revés.
* Hay otro modelo heredado que quiere añadir una restricción AND, y lo añade al final, como hemos hecho nosotros:
```python
args += [('field', operator, name)]
return super(...).name_search(..., args, ...)
```
entonces el dominio quedará mal compuesto si nuestra llamada se ejecuta antes que la del otro. Justo en este caso es más difícil que pase, puesto que nuestro modelo se declara por primera vez aquí, pero incluso así puede pasar en la v10 porque tenemos la posibilidad de heredar de `base` y que todos los modelos cojan esa modificación, quedándose la cadena de herencia como `nuestro_modelo` > `modificación en el base` > `base`, y dando este dominio, que no es correcto:

```
[('|', ('code', operator, name), ('field', operator, name), (self._rec_name, operator, name)]
```